test/forcing/ForcingsEngineLumpedDataProvider_Test.cpp

```diff
@@ -0,0 +1,125 @@
+#include "DataProviderSelectors.hpp"
+#include "ForcingsEngineDataProvider.hpp"
+#include <gtest/gtest.h>
+
+#include <NGenConfig.h>
+#if NGEN_WITH_MPI
+#include <mpi.h>
+#endif
+
+#include <forcing/ForcingsEngineLumpedDataProvider.hpp>
+#include <forcing/AorcForcing.hpp>
+#include <utilities/python/InterpreterUtil.hpp>
+
+#include <string>
+
+struct ForcingsEngineLumpedDataProviderTest
+  : public testing::Test
+{
+    using provider_type = data_access::ForcingsEngineLumpedDataProvider;
+
+    ForcingsEngineLumpedDataProviderTest()
+    {
+        #if NGEN_WITH_MPI
+        MPI_Comm_size(MPI_COMM_WORLD, &mpi_.size);
+        MPI_Comm_rank(MPI_COMM_WORLD, &mpi_.rank);
+        #endif
+    }
+
+    static constexpr const char* config_file = "data/forcing/forcings-engine/config_aorc.yml";
+    static const std::time_t time_start;
+    static const std::time_t time_end;
+    static std::shared_ptr<utils::ngenPy::InterpreterUtil> gil_;
+    static std::unique_ptr<provider_type> provider_;
+
+    struct { int initialized = 0, finalized = 0, size = 1, rank = 0; } mpi_{};
+
+    static void SetUpTestSuite();
+    static void TearDownTestSuite();
+};
+
+using TestFixture = ForcingsEngineLumpedDataProviderTest;
+
+constexpr const char* TestFixture::config_file;
+const std::time_t TestFixture::time_start = data_access::detail::parse_time("2024-01-17 01:00:00");
```

Comment:
One note that I just realized I need to change: `time_start` has no actual impact on the Forcings Engine's time domain, so while this is currently incorrect, the instance still starts at 2023-01-17 01:00:00. From what I can think of currently, there isn't an easy way to assert that the time matches, since only the time duration is exposed from the forcings engine (i.e. `get_data_start_time` and others report in seconds with an epoch of 0).

This relates back to #858.

Reply:
@jduckerOWP For the Forcings Engine in particular, could it expose a `start_time` variable via BMI that we can use to cross-check that the model engine's simulation start time matches the forcings engine's configured/available-data start time?

Reply:
Alternately, could it expose that as a pseudo-variable that we can set via BMI, rather than living in the config file?

Reply:
@PhilMiller we certainly can expose a start time via BMI for cross-checking timestamps between the Forcings Engine and the framework. Regarding @program-- comment for the timestamp discrepancy we can discuss this more in our meeting today but generally the NWM operational forecast iterations are initialize at some start time (0z,6z,12z,18z etc.) and the first available forecast data is at forecast hour 1 of the simulation. Therefore, generally hour 0 (or start time of simulation) will basically need to be forced with some "hot start" fields if models require forcings at the actual start time. For retrospective datasets however, we can modify the Forcings Engine to directly provide meteorological forcings at that start time hour we'll just need some revisions to the workflow to make that work.